src/cljc/rems/common/util.cljc

```diff
@@ -19,6 +19,166 @@
   (is (= nil
          (re-matches +phone-number-regex+ "+35845000000000000000000000000000"))))
 
+;; regex from https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp
+(def +valid-ip-address-regex+ #"(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$")
+
+(def +valid-ip-address-regex-version-six+ #"(([0-9]|[a-f]){1,4}(\:)){7}(([0-9]|[a-f]){1,4})$")
+
+;; https://stackoverflow.com/questions/2814002/private-ip-address-identifier-in-regular-expression
+;; for now, this test should test for the following private IPv4 address pattaerns
+;; 0.x.x.x
+;; 10.x.x.x - Private network, local communications
+;; 100.64.x.x–100.127.x.x - Private network, communications between a service provider and its subscribers
+;; 127.x.x.x - host, loopback addresses to the local host
+;; 169.254.x.x - Subnet, Used for link-local addresses[7] between two hosts on a single link when no IP address is otherwise specified
+;; 172.16.x.x – 172.31.x.x - Private network, local communications within a private network
+;; 192.0.0.x - Private network, IETF Protocol Assignments
+;; 192.0.2.x - Documentation, Assigned as TEST-NET-1, documentation and examples.
+;; 192.168.x.x - Private network, local communications within a private network
+;; 198.18.x.x - 198.19.x.x - Private network, benchmark testing of inter-network communications between two separate subnets
+;; 198.51.100.x - Documentation, Assigned as TEST-NET-2, documentation and examples.
+;; 203.0.113.x - Documentation, Assigned as TEST-NET-3, documentation and examples.
+;; 240.x.x.x–255.x.x.x - multicast
+(def +reserved-ip-address-range-regex+
+  (re-pattern
+   (str
+    "((0|10|127|224)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(100\\.(6[4-9]|[7-9][0-9]|1[0-1][0-7]|12[0-7])\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(169\\.254\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(172\\.(1[6-9]|2[0-9]|3[0-1])\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(192\\.0\\.(0|2?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(192\\.168\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(198\\.1[8-9]\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(198\\.51\\.100\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "(203\\.0\\.113\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((2[4-5][0-9])\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)")))
```

Comment:
suggestion: if you use `(?:regex)` for grouping instead of `(regex)`, the regex will be more efficient and `re-matches` will only return one value instead of a vector. Then you don't need to keep repeating `(first ...)` :)

docs (see non-capturing group): https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/regex/Pattern.html

Reply:
If you do not mind, Ill leave it the way it is

Reply:
Or the tests and the other code would be clearer if there were a `valid-ip-address?` function.

Instead of stuff like
```
  (is (= "0.0.0.0"
         (first (re-matches +valid-ip-address-regex+ "0.0.0.0"))))
```
To have stuff like this
```
  (is (valid-ip-address? "0.0.0.0"))
```